src/protocol/sort/reshare.rs

```diff
@@ -104,7 +108,7 @@ impl<F: Field> Reshare<F> for MaliciousContext<'_, F> {
         use crate::secret_sharing::ThisCodeIsAuthorizedToDowngradeFromMalicious;
         let rx_ctx = self.narrow(&ReshareMAC);
         let (x, rx) = try_join(
-            self.semi_honest_context().reshare(
+            self.narrow(&ReshareInput).semi_honest_context().reshare(
```

Comment:
Why is this needed? I don't think this is necessary. The caller should have narrowed (if necessary) before invoking "reshare"

Reply:
We need to reshare both x and rx at this step. We also need context to calculate accumulate_macs. self.semi_honest will consume self. Another option is to clone self before resharing x.